src/ui/Services/CourseMapper.cs

```diff
@@ -65,20 +68,22 @@ public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(ApiClie
                         }
                     }
                 ).ToList()),
-                CourseSubjects = new Collection<CourseSubject>(ucasCourseData.Subjects.Split(", ").Select(subject =>
-                new CourseSubject
-                {
-                    Subject = new Subject
-                    {
-                        Name = subject
-                    }
+                CourseSubjects = string.IsNullOrWhiteSpace(ucasCourseData.Subjects) 
```

Comment:
Can we break out nested items that is complex to variables please

ie

```
var courseSubject = DoSomthing();
var mappedCourse = new SearchAndCompare.Domain.Models.Course {CourseSubject = courseSubject};
```

creating an instance of SearchAndCompare.Domain.Models.Course seems big and painful to debug